Homework№1_Nikita/Homework№1_Nikita/Matrix.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Homework_1_Nikita
+{
+    class Matrix<T> : IMatrix<T>
+    {
+        public int Side { get; }
+
+        public T[,] matrix;
```

Comment:
better to hide implementation details that you matrix is based on two dimension array from outside making this field private. As well this will make impossible to change internal state of matrix other than using public methods.

Reply:
Извините, что на русском отвечаю, мой английский оставляет желать лучшего.

Я с вами полностью согласен, так действительно было бы лучше, но вы говорили создать массив с уникальным модификатором => пользователь сам выбирает тип массива. Так же стояла задача заполнить массив чем-либо. Так как пользователь должен заполнить массив и сам определить его тип, то для такой возможности массив matrix с модификатором доступа public. 

Если бы мы заранее знали тип массива, можно было бы заполнить его рандомными цифрами(как это сделал я) в методе класса, и вызывая этот метод формировать массив рандомных цифр. Но так как наперед мы не знаем типа этого массива, то такой возможности нету. 

Я считаю, что здесь все таки должен быть модификатор public, конкретно в этой реализации.

Reply:
Дозволь із тобою не погодитися, конкретно твоя реалізація матриці надає два публічних методи Insert та Take а це означає, що той, хто створить об'єкт матриці і вкаже конкретний тип елементів у матриці і зможе її заповнити через метод Insert, таким чином маючи даний філд публічним ти надаєш два різних способи (неявно) заповнити матрицю, а це не є добре. Приховавши даний філд та використовуючи ззовні тільки метод Insert ти можеш добитися того самого результати і при цьому приховати деталі внутрішньої реалізації (себто інкапсулювати деталі реалізації).

Reply:
Точняк. Спасибо.